src/Compilers/CSharp/Test/Emit2/Semantics/ExperimentalAttributeTests.cs

```diff
@@ -0,0 +1,1000 @@
+    class C
+    {
+        public static void M() { }
+    }
+}
+""";
+        var comp = CreateCompilation(new[] { src, experimentalAttributeSrc });
+        comp.VerifyDiagnostics(
+            // 0.cs(1,1): warning DiagID1: 'C' is for evaluation purposes only and is subject to change or removal in future updates.
+            // N.C.M();
+            Diagnostic("DiagID1", "N.C").WithArguments("N.C").WithLocation(1, 1)
+            );
+    }
+
+    [Theory, CombinatorialData]
+    public void Suppressed(bool inSource)
+    {
+        var libSrc = """
+[System.Diagnostics.CodeAnalysis.Experimental("DiagID1")]
+public class C
+{
+    public static void M() { }
+}
+""";
+
+        var src = """
+#pragma warning disable DiagID1
+C.M();
+""";
+
+        var comp = inSource
+            ? CreateCompilation(new[] { src, libSrc, experimentalAttributeSrc })
+            : CreateCompilation(src, references: new[] { CreateCompilation(new[] { libSrc, experimentalAttributeSrc }).EmitToImageReference() });
+
+        comp.VerifyDiagnostics();
+    }
+
+    [Theory, CombinatorialData]
+    public void InObsoleteMethod(bool inSource)
+    {
+        // Diagnostics for [Experimental] are not suppressed in [Obsolete] members
+        var libSrc = """
+public class C
+{
+    [System.Diagnostics.CodeAnalysis.Experimental("DiagID1")]
+    public static void M() { }
+}
+""";
+
+        var src = """
+class D
+{
+    [System.Obsolete("obsolete", true)]
+    void M2()
+    {
+        C.M();
+    }
+}
+""";
+
+        var comp = inSource
+            ? CreateCompilation(new[] { src, libSrc, experimentalAttributeSrc })
+            : CreateCompilation(src, references: new[] { CreateCompilation(new[] { libSrc, experimentalAttributeSrc }).EmitToImageReference() });
+
+        comp.VerifyDiagnostics(
+            // (6,9): warning DiagID1: 'C.M()' is for evaluation purposes only and is subject to change or removal in future updates.
+            //         C.M();
+            Diagnostic("DiagID1", "C.M()").WithArguments("C.M()").WithLocation(6, 9)
+            );
+    }
+
+    [Theory, CombinatorialData]
+    public void WithObsolete(bool inSource)
+    {
+        var libSrc = """
+[System.Obsolete("error", true)]
+[System.Diagnostics.CodeAnalysis.Experimental("DiagID1")]
+public class C
+{
+}
+""";
+
+        var src = """
+class D
+{
+    void M(C c)
+    {
+    }
+}
+""";
+
+        var comp = inSource
+            ? CreateCompilation(new[] { src, libSrc, experimentalAttributeSrc })
+            : CreateCompilation(src, references: new[] { CreateCompilation(new[] { libSrc, experimentalAttributeSrc }).EmitToImageReference() });
+
+        if (inSource)
+        {
+            comp.VerifyDiagnostics(
+                // 0.cs(3,12): warning DiagID1: 'C' is for evaluation purposes only and is subject to change or removal in future updates.
+                //     void M(C c)
+                Diagnostic("DiagID1", "C").WithArguments("C").WithLocation(3, 12)
```

Comment:
>Diagnostic("DiagID1", "C").WithArguments("C").WithLocation(3, 12) [](http://example.com/codeflow?start=16&length=65)

Should we report an obsolete error from source as well?

Reply:
This is [existing behavior](https://sharplab.io/#v2:EYLgtghglgdgNAFxFANnAJiA1AHwAIBMAjALABQ5MEYApgM4AOEAxjQAQDqs6A9gO50AdADEeAVxjoICKDxiCAsjQQQpK8gG9ybHWwDaAQQQIATlGBiENAKp0IAcxoAKbbrdHT5yzQAqEE44IQgDCKBB0dGw4bB5mFlZ+AcpCAMqmYswIUTHGcd6JgUIAojBiYNmxXgn+hYIAkjBWJgBmLOzRlfG+NcmCACI0KDT20jRwrm46Bigo/ApiKDIMQ2wAvGytKHQ0AJQAuhNseADMbNsQQ+hHBGxFAB4MNGa0jRed3mwgOZ5dh1pkbgAvuRgRQyHo8ERBAB5YB0HhDKxOABETxMPBMyLgbHSuwO4K4kn4QlEEjUsnkShUagggnuj2eNFeKHxJ2ubGCmhB5HIhDYfS5AN0eAALGwFE5gmxmDs/odQYCgA) from the Windows.Experimental attribute. 
I'll leave as-is.